### ISSUE TITLE
Words ending in "g" falsely reported as errors

### ISSUE THREAD

**Issue by chrismorison-wt** [streetsidesoftware/vscode-spell-checker#2352]
Sorry for the vague bug report, but I keep having intermittent issues with valid words ending in g being reported as errors:

<img width="560" alt="image" src="https://user-images.githubusercontent.com/104949709/202808128-147aabaf-133d-4620-a623-03d3f8b905bb.png">

I've had this issue with other words, but it only seems to happen if the word ends in "g" !?!

Using CodeSpellChecker v2.11.0

I've tried enabling debug logging, performance logging and logging to file, but I can't see any logs anywhere so I'm not sure how to provide any further information.

**Comment by chrismorison-wt** [comment]
In all these examples the false-positive errors end in g with the next compound word starting with A:

![image](https://user-images.githubusercontent.com/104949709/202824807-1b9dd49b-7185-426e-bc29-8b884ee3ff76.png)
 Although "GardeningAnimal" includes "gA" but doesn't show an error.

**Comment by Jason3S** [comment]
@chrismorison-wt,

This kind of thing normally caused by a over aggressive regular expression in `cSpell.ignoreRegExp`

Please share your configuration.

**Comment by chrismorison-wt** [comment]
As far as I'm aware I don't have any regex setup, unless there's one in the standard dictionaries I've enabled.  I'll try disabling them in groups...

The details from the extensions About / Debug page are here:
<details><summary>Settings</summary>
<p>

```
Workspace
{
  "name": "ASCP Web App",
  "workspaceFolders": [
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App",
      "name": "ASCP Web App",
      "index": 0
    }
  ],
  "textDocuments": [
    {
      "uri": "vscode-scm:git/scm0/input?rootUri%3Dfile%253A%252F%252F%252FUsers%252Fchrismorison%252FAllClear",
      "fileName": "git/scm0/input",
      "languageId": "scminput",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/frontend/src/app/components/modules/subscriptions/subscriptions.component.ts",
      "fileName": "ASCP Web App/frontend/src/app/components/modules/subscriptions/subscriptions.component.ts",
      "languageId": "typescript",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/frontend/src/app/common/static.service.ts",
      "fileName": "ASCP Web App/frontend/src/app/common/static.service.ts",
      "languageId": "typescript",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/frontend/src/app/components/modules/subscriptions/subscriptions.component.pug",
      "fileName": "ASCP Web App/frontend/src/app/components/modules/subscriptions/subscriptions.component.pug",
      "languageId": "jade",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/frontend/src/app/common/api.service.ts",
      "fileName": "ASCP Web App/frontend/src/app/common/api.service.ts",
      "languageId": "typescript",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/frontend/src/app/components/modules/subscriptions/subscriptions-customer-modal.component.ts",
      "fileName": "ASCP Web App/frontend/src/app/components/modules/subscriptions/subscriptions-customer-modal.component.ts",
      "languageId": "typescript",
      "isUntitled": false
    },
    {
      "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/.vscode/settings.json",
      "fileName": "ASCP Web App/.vscode/settings.json",
      "languageId": "jsonc",
      "isUntitled": false
    }
  ]
}
Configs
{
  "user": {
    "inherited": {
      "locales": "user",
      "languageIdsEnabled": "user"
    },
    "locales": [
      "en-GB"
    ],
    "languageIdsEnabled": [
      "asciidoc",
      "c",
      "cpp",
      "csharp",
      "css",
      "git-commit",
      "go",
      "handlebars",
      "haskell",
      "html",
      "jade",
      "java",
      "javascript",
      "javascriptreact",
      "json",
      "jsonc",
      "latex",
      "less",
      "markdown",
      "php",
      "plaintext",
      "pug",
      "python",
      "restructuredtext",
      "rust",
      "scala",
      "scss",
      "text",
      "typescript",
      "typescriptreact",
      "vue",
      "yaml",
      "yml"
    ]
  },
  "workspace": {
    "inherited": {
      "locales": "user",
      "languageIdsEnabled": "user"
    },
    "locales": [
      "en-GB"
    ],
    "languageIdsEnabled": [
      "asciidoc",
      "c",
      "cpp",
      "csharp",
      "css",
      "git-commit",
      "go",
      "handlebars",
      "haskell",
      "html",
      "jade",
      "java",
      "javascript",
      "javascriptreact",
      "json",
      "jsonc",
      "latex",
      "less",
      "markdown",
      "php",
      "plaintext",
      "pug",
      "python",
      "restructuredtext",
      "rust",
      "scala",
      "scss",
      "text",
      "typescript",
      "typescriptreact",
      "vue",
      "yaml",
      "yml"
    ]
  },
  "folder": {
    "inherited": {
      "locales": "user",
      "languageIdsEnabled": "user"
    },
    "locales": [
      "en-GB"
    ],
    "languageIdsEnabled": [
      "asciidoc",
      "c",
      "cpp",
      "csharp",
      "css",
      "git-commit",
      "go",
      "handlebars",
      "haskell",
      "html",
      "jade",
      "java",
      "javascript",
      "javascriptreact",
      "json",
      "jsonc",
      "latex",
      "less",
      "markdown",
      "php",
      "plaintext",
      "pug",
      "python",
      "restructuredtext",
      "rust",
      "scala",
      "scss",
      "text",
      "typescript",
      "typescriptreact",
      "vue",
      "yaml",
      "yml"
    ]
  },
  "file": {
    "uri": "file:///Users/chrismorison/AllClear/ASCP%20Web%20App/.vscode/settings.json",
    "fileName": "/Users/chrismorison/AllClear/ASCP Web App/.vscode/settings.json",
    "isUntitled": false,
    "languageId": "jsonc",
    "dictionaries": [
      {
        "name": "backwards-compatibility",
        "locales": [],
        "languageIds": [],
        "description": "Some words with accents that should be considered correct."
      },
      {
        "name": "aws",
        "locales": [],
        "languageIds": [],
        "description": "AWS dictionary for cspell."
      },
      {
        "name": "companies",
        "locales": [],
        "languageIds": [],
        "description": "Company names dictionary for cspell."
      },
      {
        "name": "cryptocurrencies",
        "locales": [],
        "languageIds": [],
        "description": "Cryptocurrencies dictionary for cspell."
      },
      {
        "name": "en-gb",
        "locales": [
          "en-GB"
        ],
        "languageIds": [],
        "description": "British English Dictionary"
      },
      {
        "name": "filetypes",
        "locales": [],
        "languageIds": [],
        "description": "Filetypes dictionary for cspell."
      },
      {
        "name": "node",
        "locales": [],
        "languageIds": [
          "javascript",
          "typescript",
          "javascriptreact",
          "typescriptreact",
          "mdx",
          "json",
          "jsonc"
        ],
        "description": "Node.js dictionary for cspell."
      },
      {
        "name": "npm",
        "locales": [],
        "languageIds": [
          "csharp",
          "javascript",
          "typescript",
          "javascriptreact",
          "typescriptreact",
          "mdx",
          "markdown",
          "asciidoc",
          "html",
          "pug",
          "jade",
          "php",
          "handlebars",
          "json",
          "jsonc"
        ],
        "description": "NPM dictionary for cspell."
      },
      {
        "name": "public-licenses",
        "locales": [],
        "languageIds": [],
        "description": "Common Public Licenses dictionary for cspell."
      },
      {
        "name": "softwareTerms",
        "locales": [],
        "languageIds": [],
        "description": "Software terms dictionary for cspell."
      }
    ],
    "languageEnabled": true,
    "fileEnabled": false,
    "configFiles": [],
    "fileIsExcluded": true,
    "fileIsIncluded": true,
    "fileIsInWorkspace": true,
    "excludedBy": [
      {
        "glob": ".vscode",
        "id": "VSCode-Config"
      }
    ]
  }
}
```
</p>
</details>

I also tried creating a completely new workspace with no files or `.vscode` folder, and the words above still triggered the same errors, so if it's caused by my settings it must be User settings not Workspace settings.

**Comment by chrismorison-wt** [comment]
I don't have any custom regex in my settings, so it must be coming from one of the standard dictionaries.

I've narrowed it down to the "typescript" dictionary:

<img width="871" alt="image" src="https://user-images.githubusercontent.com/104949709/202864701-c05bcb09-ec53-4ec2-8937-68da7bdb8522.png">

<img width="874" alt="image" src="https://user-images.githubusercontent.com/104949709/202864715-07dc9fc5-4659-4ba7-b771-1fccd8a8c772.png">

**Comment by Jason3S** [comment]
@chrismorison-wt,

- `!typescript` just turns off spell checking of TypeScript files.

CSpell configuration can also be found in `package.json`, `cspell.json`, or `cspell.config.json`. Would you try searching all files for: `ignoreRegExp` or `includeregExp`, these are the things that impact what is checked in a file.

**Comment by chrismorison-wt** [comment]
OK, I found some regex ignore in my VSCode user settings json file which included:
```
    "cSpell.ignoreRegExpList": [
        "[#.][\\w-]+",
        "(rel|href|to|if|else|ref)=(\"[^\"]+\"|'[^']+')",
        "src=\"[^\"]+\"",
        "&\\w+;",
        "(localStorage|router|ga|import|document|render|createHorse|console).*(;|},)",
        ":\\w+=\"",
        "\\$\\w+",
        "^[\\w\\t-]+\\(",
        "path: '.+?'"
    ],
```

The middle one was causing the issue, and if I remove `|ga` then those false-positives go away.  I've got no idea where these came from, but `createHorse` could've been a function in a project I was working on ~4 years ago so maybe these are from when I had a play with spell checking a while ago.  `ga()` is a function used with Google Analytics.  I've removed this entire line since it seems to be a bit of a hack.

Anyway, problem solved, thanks for your help.
